client/appengine_data.go

```diff
@@ -321,24 +324,115 @@ func (d *DatastreamPaginator) parseData(rawData []byte) any {
+		objectValues := []DatastreamObjectValue{}
+		data := result.Array()
+		for _, v := range data {
+			value := DatastreamObjectValue{}
+			_ = json.Unmarshal([]byte(v.Raw), &value)
+			objectValues = append(objectValues, value)
+		}
+		return objectValues
+	}
+	// if not an array, it must be an object
+	obj := result.Value().(map[string]interface{})
+	flattened, _ := flat.Flatten(obj, &flat.Options{Safe: true, Delimiter: "."})
+
+	keys := []string{}
+	for k := range flattened {
+		components := strings.Split(k, ".")
+		var theKey string
+		if len(components) > 1 {
+			theKey = strings.Join(components[:len(components)-1], ".")
+		} else {
+			theKey = k
+		}
+		keys = append(keys, theKey)
+	}
+	keys = removeDuplicateStr(keys)
+
+	rawObjectValues := gjson.GetMany(result.Raw, keys...)
+
+	ret := map[string][]DatastreamObjectValue{}
+	for i, item := range rawObjectValues {
+		values := []DatastreamObjectValue{}
+		value := DatastreamObjectValue{}
+
+		k := fmt.Sprintf("/%s", strings.ReplaceAll(keys[i], ".", "/"))
+
+		if item.IsArray() {
+			_ = json.Unmarshal([]byte(item.Raw), &values)
+			ret[k] = append(ret[k], values...)
+		} else {
+			_ = json.Unmarshal([]byte(item.Raw), &value)
+			ret[k] = append(ret[k], value)
+		}
+
+	}
+	return ret
+}
+
+func parseDatastreamWithIndividualAggregation(rawData []byte) any {
+	result := gjson.ParseBytes(rawData)
+
+	// first, we check if the complete timeseries is returned
+	individualValues := []DatastreamIndividualValue{}
+	if result.IsArray() {
+		data := result.Array()
 		for _, v := range data {
 			value := DatastreamIndividualValue{}
 			_ = json.Unmarshal([]byte(v.Raw), &value)
 			individualValues = append(individualValues, value)
 		}
 		return individualValues
 	}
-	// else, we're dealing with object aggregation (golint is now happy)
-	objectValues := []DatastreamObjectValue{}
-	for _, v := range data {
-		value := DatastreamObjectValue{}
-		_ = json.Unmarshal([]byte(v.Raw), &value)
-		objectValues = append(objectValues, value)
+
+	// if it's not a timeseries, it must be a snapshot (objects are returned)
+	obj := result.Value().(map[string]interface{})
+	flattened, _ := flat.Flatten(obj, &flat.Options{Safe: true, Delimiter: "."})
+
+	keys := []string{}
+	for k := range flattened {
+		components := strings.Split(k, ".")
+		theKey := strings.Join(components[:len(components)-1], ".")
+		keys = append(keys, theKey)
+	}
+	keys = removeDuplicateStr(keys)
+
+	rawIndividualValues := gjson.GetMany(result.Raw, keys...)
+
+	ret := map[string]DatastreamIndividualValue{}
+	for i, item := range rawIndividualValues {
+		value := DatastreamIndividualValue{}
+		_ = json.Unmarshal([]byte(item.Raw), &value)
+		k := "/" + strings.ReplaceAll(keys[i], ".", "/")
+		ret[k] = value
+	}
+	return ret
+}
+
+func removeDuplicateStr(strSlice []string) []string {
+	allKeys := make(map[string]bool)
+	list := []string{}
+	for _, item := range strSlice {
+		if _, value := allKeys[item]; !value {
+			allKeys[item] = true
```

Comment:
Since `allKeys` is a map, shouldn't it be enough to check if the key exists?

Reply:
The implementation has been updated